### ISSUE TITLE
dApp - sd-6 - Same connected dApps appear on every account

### ISSUE THREAD

**Issue by glitchminer** [status-im/status-desktop#15589]
# Bug Report

## Description

1. After a dApp is connected is appears in the dApp menu for every account, even when that account was not the one that was connected. 
![image](https://github.com/user-attachments/assets/0b8bdd49-6180-40b0-92cc-380d28f3e3e9)

---
2. When the same app is connected to multiple accounts the dApp button displays each app only once making it unclear what is connected and what the disconnect button will do.

---
3. The dApps button was not expected on the "All accounts" view but is shown - the list does not show which accounts are connected.

## Steps to reproduce
Connect to the same dApp from multiple wallet accounts

#### Expected behavior

- dApp button is shown on individual accounts
- The connected dApps shown are specific to the selected accounts
- the dApp menu is not displayed on the all accounts view

### Additional Information

- Status desktop version: Master, with dApp flag. 
- Operating System: Mac

**Comment by stefandunca** [comment]
@glitchminer I missed these requirements and never planned to add filtering of dapps based on the account selected. Could you please clarify where are these coming from to see the complexity and if should we fix it for the next release?

In the meantime I will test if the problem is also happening with different profiles which we expect should not overlapp.

**Comment by stefandunca** [comment]
> It will need confirmed with design now but Figma examples had the dapp button available on the individual account view only so there was an assumption that each account would display only dapps relevant to it.

That will be god to understand; @xAlisher must know better the intention there.

**Comment by clauxx** [comment]
I agree with @glitchminer that most likely that was the initial intention. It's a different thing though whether it's high priority to have or the user can live with it as a global connected dapps list for M1 (cc @xAlisher)? 

How complex or time consuming do you think it will be to add @stefandunca?

**Comment by stefandunca** [comment]
> How complex or time consuming do you think it will be to add @stefandunca?

I didn't deal with handling of accounts switching and filtering recently and we don't have logic to hide/manage the button so I can't say. For me it is high risk, however maybe UI team have a different view on the matter.

**Comment by stefandunca** [comment]
>  ... but I can tell you that if you have two accounts—one with a lot of funds and another just for fun—you might connect the second account to many dApps to explore and experiment

Indeed I didn't considered and AFAIK we don't have anything related to the concept of important and unimportant wallet account in the wallet. It is the first time I hear about this persona characteristics I will consider it in the future.

> I tend to agree with this. We also show unique connected dapps, right? If for instance two accounts are connected to the same dapp, would the user see two entries there or one? But even if we see two, the user wouldn't know which one to disconnect.

Yes, we only show one and on disconnect it will disconnect all the sessions. I remember we had a similar discussed and considered a limitation given that we descoped the session management. I mean we discussed the fapt that user can connect from different browsers/systems with the same dApp and different accounts and it has no way to mange or visualize this in the current requirements (descoped designs).

> IMHO this makes this issue a blocker. WDYT?

It looks like. I will tackle this next then.

Also there will be nice to have a clear description of this user story and acceptance criteria. For example should we show all connected dapps on all accounts selected (as it seems to me a way to be consistent)?

**Comment by stefandunca** [comment]
> Also there will be nice to have a clear description of this user story and acceptance criteria. For example should we show all connected dapps on all accounts selected (as it seems to me a way to be consistent)?

I see the following requirements from @xAlisher in discord channel

- All accounts page:
  - show all connected dApps
  - disconnect action disconnects dApp from all accounts
- Account page:
  - only dApps connected to opened account
  - disconnect action disconnects dApp from opened account

**Comment by clauxx** [comment]
do we have an "all accounts" page on desktop?

**Comment by stefandunca** [comment]
> do we have an "all accounts" page on desktop?

Yup, it is the default view for wallet. I mean hiding the button on that page will kind of hide the feature :)

**Comment by clauxx** [comment]
Then there we would show all active sessions
